### PR TITLE
CRA: Add compatibility for CRA v5

### PR DESCRIPTION
**Pull request by mrmckeb**
This PR makes the required updates to allow the preset to work with CRAv5.

Due to an issue with CRAv5 alpha (fixed in the next release), the latest version of `postcss-normalize` should be manually installed.

### CODE REVIEW

File: packages/preset-create-react-app/src/helpers/mergePlugins.ts [hunk 1]
```diff
@@ -11,16 +11,16 @@ export const mergePlugins = (...args: Plugin[]): Plugin[] =>
...
-          sockIntegration: 'whm',
-        },
-      });
-    }
+    //   updatedPlugin = new ReactRefreshWebpackPlugin({
+    //     overlay: {
+    //       sockIntegration: 'whm',
+    //     },
+    //   });
+    // }
```
Comment (mrmckeb) [inline packages/preset-create-react-app/src/helpers/mergePlugins.ts:24]: I'm not sure about this @shilman. We could just remove it, but I don't remember why this was added.

File: packages/preset-create-react-app/src/helpers/processCraConfig.ts [hunk 2]
```diff
@@ -61,10 +61,8 @@ export const processCraConfig = (
         ...rules,
         {
           oneOf: oneOf.map((oneOfRule: RuleSetRule): RuleSetRule => {
-            if (
-              isString(oneOfRule.loader) &&
-              /[/\\]file-loader[/\\]/.test(oneOfRule.loader)
-            ) {
+            // @ts-expect-error - This is a conflict with types bundled in Storybook.
+            if (oneOfRule.type === 'asset/resource') {
```
Comment (mrmckeb) [inline packages/preset-create-react-app/src/helpers/processCraConfig.ts:65]: In theory, we could look for both the old and new loader/type here and keep compat with the previous CRA releases, but not sure it's worth it as the preset has been stable and I can't see us adding back-compat features.

**Comment by kopach** [comment]
My storybook doesn't work with this change anymore. It builds successfully, but then on page load it never loads content. Spinner keeps spinning forever.

- I've tried to install `postcss-normalize` manually
- after that, I've got an error that `react-refresh` can't be found, so added it to the project also
- finally storybook loaded again, and I've got same never-ending loading spinner

image

Any ideas why I have this and what can I do to solve this? Or this is a bug?

**Comment by florpor** [comment]
@kopach if you're using react-scripts v4 then the changes of this pr do break storybook the way you described (probably due to [this](https://github.com/storybookjs/presets/pull/214/files#r716027342)).
make sure to use v3 of this package, and if you're initializing a new storybook project use the latest version of sb cli (currently 6.3.9) for the added [distinction between cra v5 and previous versions when setting this preset's package version](https://github.com/storybookjs/storybook/commit/155a18424797ca29535004c3e6617f9ac32677e2#diff-84ca8478d73779e9efd18b66524332bfe25776b89b363dfec8f84348d3149f34).

**Pull request merged**